elastica/experimental/connection_contact_joint/rod_rigidbody_connection.py

```diff
@@ -0,0 +1,417 @@
+            Index of first system for joint.
+        system_two : SystemType
+            System two of the joint connection.
+            Object of a child class of  either `CosseratRod` or `RigidBodyBase`.
+        index_two : int
+            Index of second system for joint.
+
+        Returns
+        -------
+
+        """
+        # Compute the position in the inertial frame of the specified point.
+        # The point is defined in the local coordinate system of system one and used to attach to the joint.
+        position_system_one = compute_position_of_point(
+            system=system_one, point=self.point_system_one, index=index_one
+        )
+        # Compute the position in the inertial frame of the specified point.
+        # The point is defined in the local coordinate system of system two and used to attach to the joint.
+        position_system_two = compute_position_of_point(
+            system=system_two, point=self.point_system_two, index=index_two
+        )
+
+        # Analogue to the positions, compute the velocities of the points in the inertial frames
+        velocity_system_one = compute_velocity_of_point(
+            system=system_one, point=self.point_system_one, index=index_one
+        )
+        velocity_system_two = compute_velocity_of_point(
+            system=system_two, point=self.point_system_two, index=index_two
+        )
+
+        # Compute the translational deviation of the point belonging to system one
+        # from the point belonging to system two
+        end_distance_vector = position_system_two - position_system_one
+
+        # Calculate norm of end_distance_vector
+        # this implementation timed: 2.48 µs ± 126 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
+        end_distance = np.sqrt(np.dot(end_distance_vector, end_distance_vector))
+
+        # Below if check is not efficient find something else
+        # We are checking if the point of `system_one` and point of `system_two` are at the same point in space
+        # If they are at the same point in space, it is a zero vector.
+        if end_distance <= Tolerance.atol():
+            normalized_end_distance_vector = np.array([0.0, 0.0, 0.0])
+        else:
+            normalized_end_distance_vector = end_distance_vector / end_distance
+
+        # Compute elastic force using a spring formulation as a linear function of the (undesired) distance between
+        # the two systems.
+        elastic_force = self.k * end_distance_vector
+
+        # Compute the velocity deviation of the point belonging to system one from the point belonging to system two
+        relative_velocity = velocity_system_two - velocity_system_one
+        # only penalize (e.g. apply opposite damping force) velocity aligned with the `normalized_end_distance_vector`
+        normal_relative_velocity = (
+            np.dot(relative_velocity, normalized_end_distance_vector)
+            * normalized_end_distance_vector
+        )
```

Comment:
I don't think we need this projection, we can directly have a resistive force along the direction of relative velocity

Reply:
@mstoelzle  updates on these?

Reply:
@bhosale2 Sorry, I missed this comment. GitHub issues are sometimes a bit messy...

I think I am the wrong person to ask about this. I literally copied this from the original 'FreeJoint' implementation. 

Out of my intuition, I don't think this projection is necessary. Please let me know if I can remove it.

Reply:
I just confirmed with @armantekinalp, you can go ahead.